### ISSUE TITLE
histogram for residuals plot appears to have incorrect x axis with denisty=True

### ISSUE THREAD

**Issue by dylanking42** [DistrictDataLabs/yellowbrick#873]
**Describe the bug**
I believe that the density argument to the histogram functionality of the residuals plot is not adjusting the x axis to a smaller range as it should.

**To Reproduce**
![image](https://user-images.githubusercontent.com/15676450/58886927-ac07be00-86b2-11e9-8bbe-aca8b15d1176.png)

```python
import matplotlib.pyplot as plt
from sklearn.datasets import make_regression
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from yellowbrick.regressor import ResidualsPlot

lr = LinearRegression()
X, y = make_regression(noise=2)

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.1
)

model = ResidualsPlot(lr, hist='density')
model.fit(X_train, y_train)
model.score(X_test, y_test)
model.poof()

```

**Dataset**
N/a

**Expected behavior**
the histogram with the argument density to appear as it would when constructed on its own, just attached to the residuals scatter. See: (notice the scale for the x axis when horizontal)
![image](https://user-images.githubusercontent.com/15676450/58887294-5aabfe80-86b3-11e9-841c-e7afde430fbd.png)

sns.jointplot(), or like this :[SO](https://stackoverflow.com/questions/37008112/matplotlib-plotting-histogram-plot-just-above-scatter-plot)

**Traceback**

```
n/a
```

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Python Version [3.7.3]
 - Yellowbrick Version [0.9.1]

**Additional context**
the frequency argument to hist works as expected:
![image](https://user-images.githubusercontent.com/15676450/58887239-41a34d80-86b3-11e9-8cf0-1aea46d61c73.png)

**Comment by dylanking42** [comment]
After experimenting a bit more, I think increasing the n_samples in make_regression() fixes it, for whatever reason.
![image](https://user-images.githubusercontent.com/15676450/58888408-57197700-86b5-11e9-9c63-b4229d498e64.png)

**Comment by naresh-bachwani** [comment]
Hello @dylanking42, 
I went through your issue and by exploring it a bit I came up with a possible solution. All the exponential terms below are approximated (so understand it as high or low). 
 
First, we need to note that the shape of the X_train is (90,100) i.e. the number of rows is less than columns. This often leads to overfitting. Hence the error on the training set is ought to be low(in your case of the order e^-13). Also due to the same reason, the error on the test set is sufficiently high(e^+1) as compared to train error.

Now the density in [plt.hist](https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.hist.html) is defined as `normalized_count = count/(bin_width*total_no_of_observation)`. In your case for train error, the bin_width is of the order of e^-14(e^-13/50) which makes the normalized count of the order e^+12. And for the test case, the normalized count is of order e^-2.

This visualizer plots histogram for both train and test errors(as you might have figured out) hence matplotlib by default chooses maximum range due to which it is not scaled to a smaller value(x_axis is of the order of e^+12).

When you increase the number of samples the overfitting decreases and train error increases. Hence bin width increases which in turn decreases the normalized count. Due to this, we observe a properly scaled axis in case of `n_samples=1000`. 
`bin_width =  (max(error)- min(error))/number of bins`
